### PR TITLE
Refactor js files

### PR DESCRIPTION
**Pull request by filiptypjeu**
95% of the .js files did the same thing over and over: Adding listeners that does a request. I therefore added a helper method doing exactly this and used in all .js files.

### CODE REVIEW

File: teknologr/members/static/js/base.js [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+ * Some of the options also allow a function (e: Element) => T to be passed, if the option value depends on for example data on the specific element. These options are:
+ * @param {String | (e: Element) => String} option.url
+ * @param {Object | (e: Element) => Object} dption.data
+ * @param {String | (e: Element) => String} option.newLocation
+ */
+const add_request_listener = options => {
+	const { element, method, url, data, confirmMessage, newLocation } = options;
+	const ELEMENT = $(element);
+
+	let TYPE = ELEMENT.get(0)?.tagName.toLowerCase() === "form" ? "submit" : "click";
```
Comment (backjonas) [inline teknologr/members/static/js/base.js]: TYPE can be const

File: teknologr/members/static/js/base.js [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
+ * @param {String} [option.confirmMessage] An optional confirm message
+ * @param {String} [option.newLocation]    The url to redirect to after the request is done (default: reload current page)
+ *
+ * Some of the options also allow a function (e: Element) => T to be passed, if the option value depends on for example data on the specific element. These options are:
+ * @param {String | (e: Element) => String} option.url
+ * @param {Object | (e: Element) => Object} dption.data
+ * @param {String | (e: Element) => String} option.newLocation
+ */
+const add_request_listener = options => {
+	const { element, method, url, data, confirmMessage, newLocation } = options;
```
Comment (backjonas) [inline teknologr/members/static/js/base.js]: Matter of preference: options could be destructured directly in the function definition as the options object isn't used anywhere

File: teknologr/members/static/js/base.js [hunk 3]
```diff
@@ -1,6 +1,59 @@
...
+	const ELEMENT = $(element);
```
Comment (backjonas) [inline teknologr/members/static/js/base.js]: Again just pesonal preference, but the variable names should probably not be all caps if they are not constants

**Review by backjonas — COMMENTED** [review]
Good job with with the refactoring, makes the js alot easier to read. Left some minor styling comments

File: teknologr/members/static/js/base.js (hunk 2 above)
Comment (filiptypjeu) [reply teknologr/members/static/js/base.js]: Yeah, I just forgot you could do that

**Review by backjonas — APPROVED** [review]

**Review by backjonas — APPROVED** [review]

**Pull request merged**